conjure-core/src/main/java/com/palantir/conjure/defs/validator/ConjureDefinitionValidator.java

```diff
@@ -505,4 +511,45 @@ private static boolean isSafetyMarker(Type type) {
             return false;
         }
     }
+
+    @com.google.errorprone.annotations.Immutable
+    private enum UniqueErrorNameValidator implements ConjureValidator<ConjureDefinition> {
+        INSTANCE;
+
+        @Override
+        public void validate(ConjureDefinition definition) {
+            ListMultimap<String, ErrorDefinition> errors = definition.getErrors().stream()
+                    .collect(Multimaps.toMultimap(
+                            errorDefinition -> errorDefinition.getNamespace().get()
+                                    + ':'
+                                    + errorDefinition.getErrorName().getName(),
+                            errorDefinition -> errorDefinition,
+                            ArrayListMultimap::create));
+
+            Map<String, Collection<ErrorDefinition>> duplicateErrors =
+                    Maps.filterValues(errors.asMap(), value -> value.size() > 1);
+            if (!duplicateErrors.isEmpty()) {
+                throw new IllegalStateException("Error names are made up of a namespace and name, "
+                        + "used to uniquely identify error types over the wire. "
+                        + "The following errors are defined multiple times:\n"
+                        + duplicateErrors.entrySet().stream()
+                                .map(entry ->
+                                        String.format("'%s': %s", entry.getKey(), describeDuplicates(entry.getValue())))
+                                .collect(Collectors.joining("\n")));
+            }
+        }
+
+        private static String describeDuplicates(Collection<ErrorDefinition> definitions) {
+            List<String> packages = definitions.stream()
+                    .map(errorDef -> errorDef.getErrorName().getPackage())
+                    .distinct()
+                    .toList();
+            if (definitions.size() == packages.size()) {
+                return packages.stream().collect(Collectors.joining(", ", "in packages: ", ""));
+            }
+            return definitions.stream()
+                    .map(ErrorDefinition::toString)
+                    .collect(Collectors.joining("\n\t", "defined %n times:\n\t", ""));
```

Comment:
Do we need to pass in the `definitions.size()` somewhere so `%n` is printed as expected? Or does `Collectors.joining` do something special internally

Reply:
hm, I don't see anything special in the docs.

```suggestion
                    .collect(Collectors.joining("\n\t", String.format("defined %d times:\n\t", definitions.size()), ""));
```

Reply:
err, I'm just dumb. fixed!